model/parts/output/income_info.go

```diff
@@ -139,8 +146,8 @@ func (ii *IncomeInfo) Update(output *Route) {
 		if !(payment.Payment.IsOriginator) {
 			ii.IncomeMap[payer] -= payment.Price
 		} else {
-			ii.Requesters[payee]++
-			ii.CostMap[payee] += payment.Price
+			ii.Requesters[payer]++
+			ii.CostMap[payer] += payment.Price
```

Comment:
Here is an example:
Orig -> f1 -> f2 -> S
if the one who pays is not originator, then it pays the price and payee gets the price. But if the one who pays is originator, then only payee should get the price. In your code payer is the one who gets the money?

Reply:
If I understand the code correctly, Requesters is the number of requests an originator has made (since 0 is regarded as non-originator) and CostMap stores the amount that originators have spent to make requests.
The income of the payee is added after this branch at line 148(155) anyway, and here we still don't take the request's price from the originators' income.

Reply:
All initial balances are zero, and with 50 originators, only 50 nodes initiate requests. If we deduct the price from originator's balance, most likely we will end up with negative balances for originators. In the experiment where all nodes pay, the chance for an originator to act as a forwarder can be small but they always pay. I think this makes negative balance for them.

Reply:
Yes, that's true. but this isn't related to this change. neither of these maps is storing their balance. These are originator-related values (number of generated requests and total cost of all generated requests - without paying)

The thing is these values are not related to payee, they're related to the payer (even though it's not actually paying)